src/compile/data.ts

```diff
@@ -263,9 +274,9 @@ export namespace source {
       sourceData.transform = [].concat(
         nullFilter.assemble(component),
         formula.assemble(component),
-        filter.assemble(component),
         bin.assemble(component),
-        timeUnit.assemble(component)
+        timeUnit.assemble(component),
+        filter.assemble(component)
```

Comment:
Do we really need filter to happen the last, because this is **very bad** for explaining the order of top-level `transform` and field inline transform.  (field transform came later, but with this change, this is no longer true.)